### PR TITLE
(maint) Add license_finder to Bolt.

### PR DESCRIPTION
**Pull request by shrug**
Using PDK project as a model, this adds the license_finder
gem and a task to the test gemset. Also adds all current
dependencies to the "approved" list, and adds a travis check
to validate dependency licenses.

### CODE REVIEW

**Comment by shrug** [comment]
This basically duplicates the setup in https://github.com/puppetlabs/pdk - Do we want to add this to Bolt? See any downside?

**Comment by MikaelSmith** [comment]
I'm not against it. Rubocop seems to be complaining about a few things on your PR.

File: Rakefile [hunk 1]
```diff
@@ -32,6 +32,11 @@ end
 desc "Run tests and style checker"
 task test: %w[spec rubocop]
 
+desc 'Check for unapproved licenses in dependencies'
+task(:license_finder) do
+  system('license_finder --decisions-file=.dependency_decisions.yml') || raise(StandardError, 'Unapproved license(s) found on dependencies')
```
Comment (mwaggett) [inline Rakefile]: Is there a way for the error to specify which dependencies/licenses are a problem?

File: .dependency_decisions.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+---
```
Comment (mwaggett) [inline .dependency_decisions.yml]: I assume this was generated by the license finder, but man, this is kind of hard to parse with human eyes..

File: Rakefile (hunk 1 above)
Comment (nicklewis) [reply Rakefile]: The output of `license_finder` includes that information, so it will be printed anyway.

File: .travis.yml [hunk 3]
```diff
@@ -4,6 +4,9 @@ cache: bundler
 rvm:
 - 2.4
 - 2.0
+env:
+  - CHECK = spec
+  - CHECK = license_manager
```
Comment (nicklewis) [inline .travis.yml]: This should be `license_finder`

Comment (nicklewis) [reply .travis.yml]: And I think this should just be added to the `bundle exec rake travisci rubocop` invocation. Currently it's trying to run the check from within the `bolt-modules/boltlib` directory rather than the top-level, and it's also running all the other tests before doing the license check (while also running them all separately).

**Comment by nicklewis** [comment]
Whoops, I accidentally selected approved instead of request changes. :)

**Comment by adreyer** [comment]
@shrug Is this still needed?
